### ISSUE TITLE
Add support for character arrays

### ISSUE THREAD

**Issue by milancurcic** [wavebitscientific/functional-fortran#11]
Adding support for character arrays would open up a plethora of functionality for manipulating strings. Given proper interfaces, the following functions should be useful for manipulating character strings: `complement`, `filter`, `fold[lrt]`, `head`, `init`, `insert`, `intersection`, `iterfold`, `last`, `map`, `reverse`, `set`, `sort`, `tail`, `union`.

**Comment by ghost** [comment]
This would simply be a matter of interpreting the character encoding for a numeric representation yes?
What route did you end up taking?

**Comment by milancurcic** [comment]
I don't think encoding is needed at all. I think the key to reusing existing implementations for numeric types, most of them operating on 1-d arrays, is to convert a string to an array of 1-character elements. I did this with transformation functions `arrstr` and `strarr` (see https://github.com/wavebitscientific/functional-fortran/tree/feature/support-strings). Only `head_char` and `init_char` implemented so far. To be honest, I had forgotten I started this as I ended up not needing it in my work. But it'd be nice to wrap it up, so thank you for the reminder! :)
